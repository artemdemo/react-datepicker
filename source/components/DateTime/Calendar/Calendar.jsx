import React from 'react';
import PropTypes from 'prop-types';
import onClickOutside from 'react-onclickoutside';
import Days from '../Days/Days';
import Time from '../Time/Time';
import Separator from '../Separator/Separator';
import TitleController from '../TitleController/TitleController';

import './Calendar.less';

export class Calendar extends React.Component {
    constructor(props) {
        super(props);
        this.handleClickOutside = this.handleClickOutside.bind(this);
    }

    handleClickOutside(e) {
        const { onClickOutside } = this.props;
        onClickOutside && onClickOutside(e);
    }

    handleDateChange(newDate) {
        const { onChange } = this.props;
        onChange && onChange(newDate);
    }

    handleTimeChange(newDate) {
        const { onChange } = this.props;
        onChange && onChange(newDate);
    }

    handleMonthChange() {}

    renderDays() {
        const { date, isValidDate } = this.props;
        return (
            <Days
                date={date}
                isValidDate={isValidDate}
                onDateChange={this.handleDateChange.bind(this)} />
        );
    }

    renderTime() {
        const { date, timeFormat } = this.props;
        if (!timeFormat) {
            return null;
        }
        return (
            <div>
                <Separator />
                <Time
                    date={date}
                    timeFormat={timeFormat}
                    onChange={this.handleTimeChange.bind(this)} />
            </div>
        );
    }

    render() {
        const { date } = this.props;
        return (
            <div className='datetime-calendar'>
                <TitleController
                    date={date}
                    format='MMMM, YYYY'
                    onChange={this.handleMonthChange.bind(this)} />
                <Separator />
                {this.renderDays()}
                {this.renderTime()}
            </div>
        );
    }
}

Calendar.displayName = 'Calendar';

Calendar.propTypes = {
    date: PropTypes.shape({}).isRequired,
    timeFormat: PropTypes.string.isRequired,
    onClickOutside: PropTypes.func,
    onChange: PropTypes.func,
    isValidDate: PropTypes.func,
};

Calendar.defaultProps = {
    timeFormat: null,
    onClickOutside: null,
    onChange: null,
    isValidDate: null,
};

export default onClickOutside(Calendar);
