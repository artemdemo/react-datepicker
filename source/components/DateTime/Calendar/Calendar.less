.datetime-calendar {
    width: 230px;
    padding: 5px;
    background-color: white;
    box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
    border-radius: 3px;
}

.datetime-calendar__view {
    height: 200px;
}
