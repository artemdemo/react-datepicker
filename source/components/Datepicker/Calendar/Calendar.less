.datepicker-calendar {
    width: 230px;
}
