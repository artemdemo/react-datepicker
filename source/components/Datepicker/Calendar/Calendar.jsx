import React from 'react';
import PropTypes from 'prop-types';
import onClickOutside from 'react-onclickoutside';
import Days from '../Days/Days';
import Time from '../Time/Time';
import Separator from '../Separator/Separator';

import './Calendar.less';

class Calendar extends React.Component {
    constructor(props) {
        super(props);
        this.handleClickOutside = this.handleClickOutside.bind(this);
    }

    handleClickOutside(e) {
        const { onClickOutside } = this.props;
        onClickOutside && onClickOutside(e);
    }

    handleDateChange(newDate) {
        const { onChange } = this.props;
        onChange && onChange(newDate);
    }

    handleTimeChange(newDate) {
        const { onChange } = this.props;
        onChange && onChange(newDate);
    }

    renderDays() {
        const { date } = this.props;
        return (
            <Days date={date} onDateChange={this.handleDateChange.bind(this)} />
        );
    }

    render() {
        const { date } = this.props;
        return (
            <div className='datepicker-calendar'>
                {this.renderDays()}
                <Separator />
                <Time date={date} onChange={this.handleTimeChange.bind(this)} />
            </div>
        );
    }
}

Calendar.propTypes = {
    onClickOutside: PropTypes.func,
    onChange: PropTypes.func,
    date: PropTypes.shape({}),
};

Calendar.defaultProps = {
    onClickOutside: null,
    onChange: null,
    value: null,
};

export default onClickOutside(Calendar);
