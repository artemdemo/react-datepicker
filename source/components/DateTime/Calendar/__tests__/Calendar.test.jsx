import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';
import { Calendar } from '../Calendar';

describe('<Calendar />', () => {
    const date = moment(1502459094000);

    it('simple render', () => {
        const tree = renderer.create(
            <Calendar
                date={date}
                timeFormat='HH:mm:ss:SSS'
            />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
