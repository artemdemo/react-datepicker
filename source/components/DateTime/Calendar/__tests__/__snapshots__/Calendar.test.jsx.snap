// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> simple render 1`] = `
<div
  className="datetime-calendar"
>
  <table
    className="datetime-title-controller"
  >
    <thead>
      <tr>
        <th
          className="datetime-title-controller-change-month datetime-title-controller-change-month_backward"
          onClick={[Function]}
        />
        <th
          className="datetime-title-controller-title datetime-title-controller-title_clickable"
          colSpan="5"
          onClick={[Function]}
        >
          August, 2017
        </th>
        <th
          className="datetime-title-controller-change-month datetime-title-controller-change-month_forward"
          onClick={[Function]}
        />
      </tr>
    </thead>
  </table>
  <hr
    className="datetime-separator"
  />
  <div
    className="datetime-calendar__view"
  >
    <table
      className="datetime-days"
    >
      <thead>
        <tr>
          <th>
            Su
          </th>
          <th>
            Mo
          </th>
          <th>
            Tu
          </th>
          <th>
            We
          </th>
          <th>
            Th
          </th>
          <th>
            Fr
          </th>
          <th>
            Sa
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              30
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              31
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              1
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              2
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              3
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              4
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              5
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              6
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              7
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              8
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              9
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              10
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_current datetime-day__content_selected"
            >
              11
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              12
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              13
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              14
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              15
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              16
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              17
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              18
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              19
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              20
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              21
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              22
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              23
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              24
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              25
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              26
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              27
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              28
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              29
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              30
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content"
            >
              31
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              1
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              2
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              3
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              4
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              5
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              6
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              7
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              8
            </div>
          </td>
          <td
            className="datetime-day"
            onClick={[Function]}
          >
            <div
              className="datetime-day__content datetime-day__content_faded"
            >
              9
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div>
    <hr
      className="datetime-separator"
    />
    <table
      className="datetime-time"
    >
      <tbody>
        <tr>
          <td>
            <div
              className="datetime-time-controller datetime-time-controller_up"
              onClick={[Function]}
            />
            <div>
              16
            </div>
            <div
              className="datetime-time-controller datetime-time-controller_down"
              onClick={[Function]}
            />
          </td>
          <td>
            :
          </td>
          <td>
            <div
              className="datetime-time-controller datetime-time-controller_up"
              onClick={[Function]}
            />
            <div>
              44
            </div>
            <div
              className="datetime-time-controller datetime-time-controller_down"
              onClick={[Function]}
            />
          </td>
          <td>
            :
          </td>
          <td>
            <div
              className="datetime-time-controller datetime-time-controller_up"
              onClick={[Function]}
            />
            <div>
              54
            </div>
            <div
              className="datetime-time-controller datetime-time-controller_down"
              onClick={[Function]}
            />
          </td>
          <td>
            :
          </td>
          <td>
            <div
              className="datetime-time-input-wrap"
            >
              <input
                className="datetime-time-input"
                onChange={[Function]}
                type="text"
                value="000"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
